src/assets/CurrencyRateController.ts

```diff
@@ -128,9 +128,15 @@ export class CurrencyRateController extends BaseController<CurrencyRateConfig, C
 	 */
 	async fetchExchangeRate(currency: string, nativeCurrency = this.activeNativeCurrency): Promise<CurrencyRateState> {
 		const json = await handleFetch(this.getPricingURL(currency, nativeCurrency));
+		const conversionRate = Number(json[currency.toUpperCase()]);
+
+		if (!Number.isFinite(conversionRate)) {
+			throw new Error(`Invalid response for ${currency.toUpperCase()}: ${json[currency.toUpperCase()]}`);
```

Comment:
I see a potential bug throwing an error here. While ago we had an issue with concurrency in calls to the API happening asynchronously in a bad way. the solution was to add a semaphore to take care of the current state of the controller while doing calls. So, this method is used here
https://github.com/MetaMask/gaba/blob/2dc732ff1b13be27fc5f83a197f379fb8669be18/src/assets/CurrencyRateController.ts#L144
and is being called from other places with `safelyExecute` so throwing an error is fine. But throwing an error without releasing 
https://github.com/MetaMask/gaba/blob/2dc732ff1b13be27fc5f83a197f379fb8669be18/src/assets/CurrencyRateController.ts#L148
could block this method.

Reply:
WDYT?

Reply:
Good point, though this seems to be a pre-existing problem (one I introduced a while back 😬 ). `handleFetch` already throws an error in some cases, so it could already block.

I guess the solution would be to put a `try` block around the code between the `mutex.aquire()` and` releaseLock()`, and put `releaseLock()` in the `finally` clause.

Reply:
You're right, that's why we have this method called `safelyExecute` to handle errors, all async methods are called with that if it throws it just keeps polling.
For the same reason having a try catch inside safelyExecute doesn't look good from my point of view. As you mentioned before, going with 0 as value would be good instead of throwing.

Reply:
It would solve the problem you identified (the mutex never being released). It should be perfectly fine. 🤷‍♂ Whether it's being called from `safelyExecute` or not shouldn't make any difference.

Though it was `try ... finally` that I suggested, not `try ... catch`. `try ... finally` would let us release the lock in the `finally` block regardless whether an error was thrown or not, then it would let the error continue to bubble upwards if it was thrown, to be caught by `safelyExecute` or wherever else. It's a pretty standard way of dealing with cases like this where you need to ensure something is released.

Reply:
` It's a pretty standard way of dealing with cases like this` yes, I know. 
I'm fine whatever handles the issue I brought 👍

Reply:
I have created a separate PR to address this problem: #195